x509/cert.go

```diff
@@ -30,3 +33,35 @@ func ParseCertificatePEM(data []byte) ([]*x509.Certificate, error) {
 	}
 	return certs, nil
 }
+
+// ValidateCertChain takes an ordered certificate chain and validates issuance from leaf to root
+func ValidateCertChain(certChain []*x509.Certificate) error {
```

Comment:
I'd rename this to `ValidateSigningCertChain` if this is going to be distinct from TSA cert chain validation.